File: server/welcome.py
import huggingface_hub
from flask import Blueprint
from server.helper import get_body, get_field


welcome_blueprint = Blueprint("welcome", __name__)

@welcome_blueprint.route("/welcome/login", methods=[ "POST" ])
def login():
    response = dict()
    try:
        data = get_body()
        token = get_field(data, "token", str)
        huggingface_hub.login(token)
        response["success"] = True
    except:
        response["success"] = False
    finally:
        return response

@welcome_blueprint.route("/welcome/logged_in", methods=[ "POST" ])
def logged_in():
    response = dict()
    try:
        huggingface_hub.whoami()
        response["success"] = True
    except:
        response["success"] = False
    finally:
        return response

File: app/api/welcome.ts
"use server";

export async function loggedIn(): Promise<boolean> {
    const response = await fetch("http://127.0.0.1:2778/welcome/logged_in", {
        method: "POST"
    });
    const json = await response.json();
    return json?.success === true;
}

export async function login(token: string): Promise<boolean> {
    const response = await fetch("http://127.0.0.1:2778/welcome/login", {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({
            token: token
        })
    });
    const json = await response.json();
    return json?.success === true;
}